### PR TITLE
Task 1

### PR DESCRIPTION
**Pull request by jelenanemcic**

### CODE REVIEW

File: homework_4/bets_api/internal/domain/services/betService.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// NewBetService creates a new instance of BetService.
+func NewBetService(repository sqlite.BetRepository) *BetService {
+	return &BetService{
+		betRepository: repository,
+	}
+}
+
+//GetBetById returns a bet with the specific id.
+func (e BetService) GetBetById(ctx *gin.Context) (models.BetReduced, bool, error) {
+	id := ctx.Param("id")
```
Comment (vpintaric-axilis) [inline homework_4/bets_api/internal/domain/services/betService.go]: `BetService` should contain business logic only but by accepting `*gin.Context` it got coupled with `gin` and it "knows" it's being used as part of the HTTP API.

I would suggest moving `id := ctx.Param("id")` into the controller logic and changing this method to:
`func (e BetService) GetBetById(ctx context.Context, id string) (models.BetReduced, bool, error)`

Similar for other methods as well.

File: homework_4/bets_api/internal/domain/models/betReduced.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+// BetReduced represents a domain model representation of a bet without CustomerId.
+type BetReduced struct {
+	Id                   string
+	Status               string
+	SelectionId          string
+	SelectionCoefficient float64
+	Payment              float64
+	Payout               float64
+}
```
Comment (vpintaric-axilis) [inline homework_4/bets_api/internal/domain/models/betReduced.go:11]: Why is this necessary?

File: homework_4/bets_api/internal/infrastructure/sqlite/betRepository.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+	var domainBets []domainmodels.BetReduced
+	for _, storageBet := range storageBets {
+		domainBets = append(domainBets, r.betMapper.MapStorageBetToDomainBetReduced(storageBet))
+	}
+
+	return domainBets, nil
+}
+
+func (r *BetRepository) queryGetBetsByStatus(ctx context.Context, status string) ([]storagemodels.Bet, error) {
+	row, err := r.dbExecutor.QueryContext(ctx, "SELECT * FROM bets WHERE status='"+status+"';")
```
Comment (vpintaric-axilis) [inline homework_4/bets_api/internal/infrastructure/sqlite/betRepository.go]: How about this?

```suggestion
	row, err := r.dbExecutor.QueryContext(ctx, "SELECT * FROM bets WHERE status=?;", status)
```

File: homework_4/bets_api/internal/domain/models/betReduced.go (hunk 2 above)
Comment (jelenanemcic) [reply homework_4/bets_api/internal/domain/models/betReduced.go:11]: Because we do not want to return CustomerId. Maybe there is a better way to do so, but I could not think of any.

File: homework_4/bets_api/internal/domain/services/betService.go (hunk 1 above)
Comment (jelenanemcic) [reply homework_4/bets_api/internal/domain/services/betService.go]: I fixed it :)

File: homework_4/bets_api/internal/infrastructure/sqlite/betRepository.go [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+	if err != nil {
+		return domainmodels.BetReduced{}, false, errors.Wrap(err, "bet repository failed to get a bet with id: "+id)
+	}
+
+	domainBet := r.betMapper.MapStorageBetToDomainBetReduced(storageBet)
+	return domainBet, true, nil
+}
+
+func (r *BetRepository) queryGetBetByID(ctx context.Context, id string) (storagemodels.Bet, error) {
+	row, err := r.dbExecutor.QueryContext(ctx, "SELECT * FROM bets WHERE id='"+id+"';")
```
Comment (vpintaric-axilis) [inline homework_4/bets_api/internal/infrastructure/sqlite/betRepository.go]: You still have a couple of these where you're not using placeholders+arguments in the queries.

```suggestion
	row, err := r.dbExecutor.QueryContext(ctx, "SELECT * FROM bets WHERE id=?;", id)
```

File: homework_4/bets_api/internal/domain/models/betReduced.go (hunk 2 above)
Comment (vpintaric-axilis) [reply homework_4/bets_api/internal/domain/models/betReduced.go:11]: We don't _need_ to or we _should not_ return CustomerId?

I believe it's the case of the former so there's no reason to overcomplicate things. Just returning the model with customer ID is fine 🤷

Comment (jelenanemcic) [reply homework_4/bets_api/internal/domain/models/betReduced.go:11]: I understood that we must not return CustomerId because of GDPR (something like that was mentioned in the class).

Comment (vpintaric-axilis) [reply homework_4/bets_api/internal/domain/models/betReduced.go:11]: Ah, alright, valid reason. It wasn't necessary to handle this in the homework though 😄 .

A cleaner way to do this would be to have a `BetHttpDto` model (or something similarly named) instead of `BetReduced` which only has the data we wish to return to the client, have controller receive full `Bet` model from the `BetService` and then map `Bet` -> `BetHttpDto` (possibly have a whole separate mapper class to do this).

That way you don't need to drag through the `BetReduced` model through multiple layers of the application and you have this "what do we want to return to the client" logic in one single place where it can be easily tested.

**Review by vpintaric-axilis — APPROVED** [review]
